device/components/RTA_OnDeviceComponent.brs

```diff
@@ -126,6 +125,22 @@ function processObserveFieldRequest(request as Object) as Dynamic
 	end if
 
 	field = args.field
+
+	' If match was provided, check to see if it already matches the expected value
+	match = args.match
+	if isAA(match) then
+		result = processGetValueAtKeyPathRequest(match)
+		if result.found <> true then
+			return buildErrorResponseObject("Match was requested and key path was not valid")
+		end if
+		if result.value = args.match.value then
```

Comment:
very minor we can use node = match.value since we are creating local values

Reply:
node comes from keyPath minus the final component. match comes from match.keyPath